tools/publisher/src/subcommand/upgrade_runtime_crates_version.rs

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use crate::fs::Fs;
+use anyhow::{anyhow, bail, Context};
+use clap::Parser;
+use regex::Regex;
+use std::path::{Path, PathBuf};
+
+#[derive(Parser, Debug)]
+pub struct UpgradeRuntimeCratesVersionArgs {
+    /// The version of runtime crates you want the code generator to use (e.g. `0.52.0`).
+    #[clap(long)]
+    version: String,
+    /// The path to the `gradle.properties` file. It will default to `gradle.properties` if
+    /// left unspecified.
+    #[clap(long, default_value = "gradle.properties")]
+    gradle_properties_path: PathBuf,
```

Comment:
Is there a use case for performing this substitution in another file? Unsure this param is needed.

Reply:
No, the parameter is mostly there to allow invoking the script from a difference directory (i.e. avoid assuming that the file exists in the current working directory).